gossipd/gossipd.c

```diff
@@ -1043,201 +953,191 @@ static u8 *create_channel_update(const tal_t *ctx,
 	}
 
 	msg = wire_sync_read(tmpctx, HSM_FD);
-	if (!msg || !fromwire_hsm_cupdate_sig_reply(ctx, msg, &update)) {
+	if (!msg || !fromwire_hsm_cupdate_sig_reply(NULL, msg, &update)) {
 		status_failed(STATUS_FAIL_HSM_IO,
 			      "Reading cupdate_sig_req: %s",
 			      strerror(errno));
 	}
 
-	return update;
+	/* We always tell peer, even if it's not public yet */
+	if (!is_chan_public(chan)) {
+		struct peer *peer = find_peer(daemon,
+					      &chan->nodes[!direction]->id);
+		if (peer)
+			queue_peer_msg(peer, update);
+	}
+
+	msg = handle_channel_update(daemon->rstate, take(update), caller);
+	if (msg)
+		status_failed(STATUS_FAIL_INTERNAL_ERROR,
+			      "%s: rejected local channel update %s: %s",
+			      caller,
+			      /* This works because handle_channel_update
+			       * only steals onto tmpctx */
```

Comment:
what do you mean by 'steals onto'?

Reply:
The line above says take(update), so we should not generally touch it (handle_channel_update owns it, is allowed to free it).  In fact, this is a "shouldnt happen" path already, and handle_channel_update actually just tal_steal(tmpctx) the update in the error path, rather than actually freeing it...

Reply:
Sweet. That makes sense. You could roughly translate it as 'steal onto tmpctx' -> `tal_steal(tmpctx)`